includes/spoolchanges.js

```diff
@@ -71,18 +67,23 @@ module.exports = function(dbUrl, log, bufferSize, callback) {
     url: dbUrl + '/_changes',
     qs: { seq_interval: 10000 }
   };
-  client(r)
-    .pipe(liner())
+  var c = client(r);
+  c.end();
+
+  c.on('response', function(resp) {
+    if (resp.statusCode !== 200) {
+      c.abort();
+      callback(new error.BackupError('SpoolChangesError', 'Changes request failed with status code ' + resp.statusCode));
+    } else {
+      console.error('Streaming changes to disk:');
+    }
+  }).pipe(liner())
     .pipe(change(onChange))
     .on('finish', function() {
       processBuffer(true);
       logStream.write(':changes_complete ' + lastSeq + '\n');
       logStream.end();
       console.error('');
-      if (doccount === 0) {
-        callback(new Error('zero documents found - nothing to do'), null);
-      } else {
-        callback(null, {batches: batch, docs: doccount});
-      }
+      callback();
```

Comment:
Nothing is looking at the err in that callback in master currently. It just gets ignored. So these changes don't affect behaviour.
I don't think having 0 docs in your db is an error anyway. So I'm suggesting we get rid of the `doccount` check.

_Backing up an empty db:_
```
$ couchbackup --url https://samsmith.cloudant.com --db newdb1
================================================================================
Performing backup on https://****:****@samsmith.cloudant.com/newdb1 using configuration:
{
  "bufferSize": 500,
  "log": "/var/folders/5p/11lmsvwn5l91cn7ysrs4nm1r0000gn/T/tmp-34812WgkPowX4hEdx.tmp",
  "mode": "full",
  "parallelism": 5
}
================================================================================
Streaming changes to disk:

couchbackup:backup finished { total: 0 } +0ms
$ echo $?
0
```